### ISSUE TITLE
Load pretrained model with new load_parameters() API

### ISSUE THREAD

**Issue by ShownX** [apache/incubator-mxnet#12334]
## Description
Cannot load only convolutional layers with pre-trained model using ```load_parameters()``` API 

## Environment info (Required)
Ubuntu 16.04
mxnet-cu90 1.3.0b20180821

## Error Message:
missing_parameters

## Minimum reproducible example
```
from mxnet.gluon.model_zoo.vision import get_model

# download the pretrained model with the path: model_path
model = get_model('resnet101_v2')
model.features.load_parameters(model_path)
```

## What have you tried to solve it?
The reason of this issue is load_parameters called from ```features block```, in which name of parameters starts with numbers. But the pretrained model parameters starts with ```features```.
The ```load_parameters``` cannot find the corresponding name in this case.

One way to work around this issue is using the ```model.save_params()``` to save again and load using the API ```load_parameters```. However, the save_params() is deprecated.

To prevent any problem with loading parameters, I propose this issue and hope the team can fix it.

**Comment by ankkhedia** [comment]
@mxnet-label-bot [Gluon, Model Zoo, Bug]

**Comment by L1aoXingyu** [comment]
```
from mxnet.gluon.model_zoo.vision import get_model

# download the pretrained model with the path: model_path
model = get_model('resnet101_v2')
model.features.load_parameters(model_path, allow_missing=True, ignore_extra=True)
```
This can work.

**Comment by ShownX** [comment]
Sometimes, we need to initialize using the other models weights, not only the weights stored on the server.

I proposed this issue for future use since the ```save_params()``` is deprecated in the future. It should be raised because it would be super user-unfriendly.

**Comment by zhreshold** [comment]
After transition from `save_params` to `save_parameters` it is no longer enforcing prefix check, and it's really error prone if we are going to load partial weights from different models.  
So `save_parameters` and `load_parameters` are seriously only for serialization IMO. 

As suggested by @L1aoXingyu I think it's safer to copy loaded parameters between each other, at least you will know what parameters are indeed copied. 

Let me know what's your concern if directly loading partial weights is not applicable. @ShownX

**Comment by ShownX** [comment]
Right now, I meet the issue like:
1. I have a pre-trained model, e.g. resnet trained on my own datasets. Assume it is saved by the function ```save_parameters()```. The key of the saved function would be the format such as ```features.0.0.mean```

2.1. If I just use exactly model but only initialize the features using the pretrained model, the previous function does not work: ```model.features.initialize(model_path)```. 
2.2. Based on this pretrained model, I would like to expand the functionalities or modify the model based on the pre-trained model. Let's say I have the several ```hyridizesequences``` such as ```features0, features1, ...```, which change the structure of the models and the key.

3. In both cases, the initialization would fail because it cannot find the corresponding key from the ndarray formated pre-trained model and the modified pre-trained models.

For example, ndarray pretrained model is ```features.0.0.mean```
but the corresponding key of parameters in the case 2.1 is ```0.0.mean``` and the key of parameters in case 2.2 might be ```1.0.mean```

However, in the deprecated function ```save_params()```, which uses the layer name as the key, I can find the corresponding key in both cases.

Hope the mxnet team can address this issue. Otherwise, it would be the terrible user experience when the users face both cases.

**Comment by ShownX** [comment]
Thanks. Right now, I can use ```save_params()``` to get what I want. But it is deprecated in the future. I want to address this issue when the mxnet team actually decide to deprecate this function and hope can update the functions ```save_parameters()``` or ```load_parameters()```.

There are two cases I mention before. 1. Training the custom dataset with the pre-trained model; 2. Modifying the backbone network such as ResNet and design a new network.

**Comment by zhreshold** [comment]
@piiswrong Do you have some comments here?

**Comment by piyushghai** [comment]
@ShownX If the issue is resolved for you, can you close this issue ? Thanks :)

**Comment by ShownX** [comment]
I do not think it is resolved.
